src/Microsoft.AspNet.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -27,7 +27,7 @@ protected virtual void HandleNonHttpsRequest(AuthorizationContext filterContext)
             // body correctly.
             if (!string.Equals(filterContext.HttpContext.Request.Method, "GET", StringComparison.OrdinalIgnoreCase))
             {
-                filterContext.Result = new HttpStatusCodeResult(StatusCodes.Status403Forbidden);
+                filterContext.Result = new HttpStatusCodeResult(StatusCodes.Status405MethodNotAllowed);
```

Comment:
Choosing between HTTP 4xx status codes is tricky. Is the problem the protocol, the method, the URL, the body, a header?

In this particular case my impression is that the issue isn't about using the wrong HTTP *method*. It's that the wrong *protocol* is being used to access the resource. That is, for someone seeing this error the fix isn't to change from POST to GET, it's to change from HTTP to HTTPS.